src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -1261,6 +1667,172 @@ void MVN::MVNRefExecutor::mvn_ref(const uint8_t* src_data, uint8_t* dst_data) {
+    parallel_for(N, [&](size_t b) {
+        std::vector<float> mean_buffer(aux_buffer_size * threads_num);
+        std::vector<float> variance_buffer(aux_buffer_size * threads_num);
+        size_t b_offset = b * C * D * H * W;
+        if (mvnAttrs.execAcrossChannels_) {
+            float size_inv = 1.f / static_cast<float>(C * D * H * W);
+            parallel_nt(0, [&](const int ithr, const int nthr) {
+                size_t start = 0, end = 0;
+                splitter(D * H * W, nthr, ithr, start, end);
+
+                auto arg = jit_mvn_call_args();
+                arg.src = src_data + (b_offset + (start * C)) * src_data_size;
+                arg.sum = &mean_buffer[aux_buffer_size * ithr];
+                arg.work_amount = static_cast<size_t>((end - start) * C);
+                (*mvn_mean_kernel)(&arg);
+            });
+            for (size_t i = 1; i < threads_num; i++) {
+                mean_buffer[0] += mean_buffer[i];
+            }
+            mean_buffer[0] *= size_inv;
+            if (mvnAttrs.normalizeVariance_) {
+                parallel_nt(0, [&](const int ithr, const int nthr) {
+                    size_t start = 0, end = 0;
+                    splitter(D * H * W, nthr, ithr, start, end);
+
+                    auto arg = jit_mvn_call_args();
+                    arg.src = src_data + (b_offset + (start * C)) * src_data_size;
+                    arg.mean = &mean_buffer[0];
+                    arg.variance = &variance_buffer[aux_buffer_size * ithr];
+                    arg.work_amount = static_cast<size_t>((end - start) * C);
+                    (*mvn_variance_kernel)(&arg);
+                });
+
+                for (size_t i = 1; i < threads_num; i++) {
+                    variance_buffer[0] += variance_buffer[i];
+                }
+                if (mvnAttrs.epsMode_ == INSIDE_SQRT)
+                    variance_buffer[0] = 1.f / sqrtf(variance_buffer[0] * size_inv + mvnAttrs.epsValue_);
+                else if (mvnAttrs.epsMode_ == OUTSIDE_SQRT)
+                    variance_buffer[0] = 1.f / (sqrtf(variance_buffer[0] * size_inv) + mvnAttrs.epsValue_);
+
+                parallel_nt(0, [&](const int ithr, const int nthr) {
+                    size_t start = 0, end = 0;
+                    splitter(D * H * W, nthr, ithr, start, end);
+
+                    auto arg = jit_mvn_call_args();
+                    arg.src = src_data + (b_offset + (start * C)) * src_data_size;
+                    arg.dst = dst_data + (b_offset + (start * C)) * dst_data_size;
+                    arg.mean = &mean_buffer[0];
+                    arg.variance = &variance_buffer[0];
+                    arg.work_amount = static_cast<size_t>(end - start);
+                    arg.oc_off = 0;
+                    arg.post_op_data = post_ops_data_;
+                    (*mvn_kernel)(&arg);
+                });
+            } else {
+                parallel_nt(0, [&](const int ithr, const int nthr) {
```

Comment:
This code block seems to be duplicated in several places. Can we make it a function to be reused.

Reply:
extracted, thanks.